signer/src/storage/in_memory.rs

```diff
@@ -237,7 +257,7 @@ impl Store {
         let context_window_end_block = std::iter::successors(first_block, |block| {
             Some(self.bitcoin_blocks.get(&block.parent_hash).unwrap_or(block))
         })
-        .nth(context_window as usize);
+        .nth(context_window as usize - 1);
```

Comment:
I'm having troubles understanding why this is correct: in `get_pending_withdrawal_requests` given a `context_window` value we end up with that many + 1 blocks in `extended_context_window`, so it seems here we would consider one less block.

Reply:
Oh, I was trying stuff out when one of the tests was failing and I forgot to remove this.

Reply:
https://github.com/stacks-network/sbtc/pull/1559/commits/5ee090603fc9afaf23011c9ccdc9e851c86da74f